### ISSUE TITLE
Do you need help?

### ISSUE THREAD

**Issue by qhga** [hughesjs/OpenEQiSports#1]
Cool to see that you start this project. I got mine yesterday and was immediately looking for an alterative to the app. If I can help you with anything (e.g. collecting traces for my device ‎CS-WP8, debugging, documentation, etc.) just let me know.

**Comment by hughesjs** [comment]
I will happily accept any help going mate. Have you got a Mac/iPad? If so, could you use nRF Connect to hit your treadmill with the commands detailed in the protocol readme and see if it lines up? If it does that'll be a lot easier than needing an adapter for every model, but it'd be good to know up-front.

I'm planning on starting work with a PoC app this weekend then plan on moving on to a more fully functional app, probably using MAUI but maybe Flutter.

**Comment by hughesjs** [comment]
You might be interested to know I now have a demo app that can start, stop, and increment the speed. #3

**Comment by hughesjs** [comment]
@qhga - Could you please try the proof of concept app I just merged to confirm what's in your PR? It's MAUI so should "just work" cross platform, but I've only tested it on iOS.

**Comment by qhga** [comment]
@hughesjs I tried to build it on my system (NixOS) which has some problems to use local workloads (maui-android) so I now try to build it inside a Ubuntu Docker container which... might also fail (I never used C#). So if it is possible that you provide me with an .apk I can try it out with my model.

**Comment by qhga** [comment]
Wait, got it working inside the container. Trying it now!

**Comment by qhga** [comment]
The app does not seem to work properly. But I found this [potential maui bug](https://github.com/dotnet/maui/discussions/9510) while trying to debug the app via adb and since I also removed the ios/mac lines from the csproj file since I got this error it could be the same problem as suspected in the discussion:
```
/opt/bitnami/dotnet-sdk/bin/sdk/8.0.101/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1178: The project depends on the following workload packs that do not exist in any of the workloads available in this installation: Microsoft.ios.Sdk.net7 [/app/EqiPoc.csproj::TargetFramework=net7.0-ios]
/opt/bitnami/dotnet-sdk/bin/sdk/8.0.101/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1178: You may need to build the project on another operating system or architecture, or update the .NET SDK. [/app/EqiPoc.csproj::TargetFramework=net7.0-ios]
```

Could also be the android sdk version I used, idk... Also not familiar with android development. (I used build-tools & platform android-33 - since I use Android 13)

I am out of ideas for now and cannot debug more right now. I might be able to spend some more hours on Sunday. Hope I could help. I also added the Dockerfile which I used to build the app to my fork/branch if you want to investigate further.

**Comment by hughesjs** [comment]
I'll build an APK later today and set up a 0.1 release

**Comment by hughesjs** [comment]
I've just checked, I actually have the same model as you, so all should be fine

**Comment by qhga** [comment]
> https://github.com/hughesjs/OpenEQiSports/releases/tag/v0.0.1 <--- Try that

That one does not even install and gives the error: `App not installed as package appears to be invalid` when copying the apk to the phone and `adb: failed to install com.hughesjs.eqipoc.apk: Failure [INSTALL_PARSE_FAILED_NO_CERTIFICATES: Failed to collect certificates from /data/app/vmdl321117973.tmp/base.apk: Attempt to get length of null array]` while installing via adb.

I got the same error yesterday with the unsigned version. Would be great if you could me provide the signed apk

**Comment by hughesjs** [comment]
Yep... Didn't realise APKs needed to be signed...

**Comment by hughesjs** [comment]
I've updated the linked release.

Are you happy connecting on Discord or something? This issue is probably not the most efficient way of communicating if you're wanting to keep contributing

**Comment by qhga** [comment]
I installed the new release but I get the same behaviour as with my build: App opens for a second, shows purple/blueish background and flashes .NET, then exits. So it seems like the docker build process I set up works but the apk does not. Weird, I would also expect that MAUI produces working builds for all platforms not just IOS...

Sure we can connect on Discord! Should I send you my handle to the mail address that is shown on your profile?
